metricbeat/module/docker/cpu/helper.go

```diff
@@ -80,27 +80,30 @@ func getNewCpu(stats *dc.Stats) CPURaw {
 
 func (c *CPUService) PerCpuUsage(stats *dc.Stats) common.MapStr {
 	var output common.MapStr
-	if cap(getNewCpu(stats).PerCpuUsage) == cap(getOLdCpu(stats).PerCpuUsage) {
+	if cap(getNewCpu(stats).PerCpuUsage) == cap(getOldCpu(stats).PerCpuUsage) {
```

Comment:
@andrewkroh, Theoretically & technically you're asolutely right! 
But the types i'm comparing are both arrays ( uint64  arrays ) and the documentation of the moethod cap(v Type) says that this function returns the capacity of v according to it's type. so when it's about arrays, it returns the number of elements in v, same as len(v). So the probability to have  an index out of range runtime error tend towards zero! correct me if i'm wrong !
https://golang.org/pkg/builtin/#cap